src/components/SearchBar/SearchBar.component.jsx

```diff
@@ -0,0 +1,60 @@
+import React, { useState } from 'react';
+
+import { useHistory } from 'react-router-dom';
+// import styled from 'styled-components';
+
+import './SearchBar.styles.css';
+
+export default function SearchBar({ initSearchQuery, onChange, homepage }) {
+  const [searchQuery, setSearchQuery] = useState(initSearchQuery);
+  // eslint-disable-next-line prefer-const
+  let history = useHistory();
+
+  const handleChange = (event) => {
+    setSearchQuery(event.target.value);
+  };
+  const triggerChange = () => {
+    if (searchQuery) {
+      onChange(searchQuery);
+      const searchRedirect = `/search/${searchQuery}`;
+      history.push(searchRedirect);
+    }
+  };
+
+  const handleKeyDown = (event) => {
+    if (event.key === 'Enter') {
+      triggerChange();
+    }
+  };
+
+  let InputSearchBar;
+  if (homepage) {
+    InputSearchBar = (
+      <div className="search_homepage">
+        <input
+          type="text"
+          className="full_search_input"
+          placeholder="eg. React Tutorials"
+          value={searchQuery}
+          onChange={handleChange}
+          onKeyDown={handleKeyDown}
+        />
+      </div>
+    );
+  } else {
+    InputSearchBar = (
+      <div className="search">
+        <input
+          type="text"
+          className="search_input"
+          placeholder="Search"
+          value={searchQuery}
+          onChange={handleChange}
+          onKeyDown={handleKeyDown}
+        />
+      </div>
+    );
+  }
+
+  return InputSearchBar;
```

Comment:
We could check the boolean for setting the correct class names to remove code duplication:
```js
return (
      <div className={isHomepage ? 'search_homepage': 'search'}>
        <input
          type="text"
          className={isHomepage ? 'full_search_input' : ' search_input'}
          placeholder={isHomepage ? 'eg. React Tutorials' : 'Search'}
          value={searchQuery}
          onChange={handleChange}
          onKeyDown={handleKeyDown}
        />
      </div>
    )
```

Reply:
Got it.